TiddlyWiki/TiddlyWiki
# release 2.9.3

This PR is created to encourage release of 2.9.3. Main todos before release:
* [x] introduce auto checking updates availability (for now disabled by default)
  * for now, in a plugin (see comments); in 2.9.4, this will get into the core
* [x] bump version (Version.js, package.json), tag commit

Pre-releasing:
- [x] gather changelog
  - [x] summarize changes and add thanks (at least some of them)
- [x] build core, announce and attach it, the plugin for autochecking upgrades, changes summary
- [x] beta-test
  - at least 4 users tested and provided feedback

After this is merged:
- [ ] attach built files to GH release
- [ ] update site (index core, test.html, download, archive, may be content of index – what's new etc)
- [ ] announce the release

After the release:
* summarize the current release process, automate where possible (ideally: CI/CD on merge into master/main)
* update docs on dev/contribute & release process (now `dev` branch is the main branch for development; `master` is the release branch which may be renamed into `main`; all PRs should go to `dev`)

Things that I would like to add to the release very much, but better go to the next one:
* contemporary core theme (not default styles update, but a theme as a tiddler/shadow)
* deal with the "security" issues reported by github (those are in fact issues of build tools and not TW itself)
* try to substitute jQuery in repo with jQuery as a package dependency (for easy version bumps/experiments)
* finish the first wave of codestyle updating (`npm run lint` should show no errors, but I'm updating files one by one since there are more issues and linting rules themselves are not finished)
* make core upgrading work in Timimi (without need to download, import & delete)
